Tasks/ArchiveFiles/archivefiles.ts

```diff
@@ -1,7 +1,7 @@
 import path = require('path');
 import tl = require('vsts-task-lib/task');
 
-var repoRoot: string = tl.getVariable('build.sourcesDirectory');
+var repoRoot: string = tl.getVariable('System.DefaultWorkingDirectory');
```

Comment:
tl.getVariable('build.sourceDirectory') is legacy for the old node or windows agent, we can remove it for new versions of VSTS and TFS.